vllm/model_executor/models/deepseek_v2.py

```diff
@@ -180,8 +227,17 @@ def __init__(
     def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
         num_tokens, hidden_dim = hidden_states.shape
         hidden_states = hidden_states.view(-1, hidden_dim)
+
+        # Chunk the hidden states so they aren't replicated across TP ranks.
```

Comment:
Am I understanding this correctly:
- Attn does all reduce so inputs in the tp group are replicated on each rank in the tp ground
- We do the MoE with seq parallel to avoid duplication

If so, I think a comment like this would be useful just because it took me a couple minutes to catch myself up to what was going on here

Reply:
That's right - I'll improve the documentation